homeassistant/components/sensor/deconz.py

```diff
@@ -0,0 +1,196 @@
+        """Return the icon to use in the frontend."""
+        return self._sensor.sensor_icon
+
+    @property
+    def unit_of_measurement(self):
+        """Unit of measurement of this sensor."""
+        return self._sensor.sensor_unit
+
+    @property
+    def available(self):
+        """Return True if sensor is available."""
+        return self._sensor.reachable
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the sensor."""
+        attr = {
+            ATTR_BATTERY_LEVEL: self._sensor.battery,
+        }
+        return attr
+
+
+class DeconzBattery(Entity):
+    """Battery class for when a device is only represented as an event."""
+
+    def __init__(self, device):
+        """Register dispatcher callback for update of battery state."""
+        self._device = device
+        self._name = self._device.name + ' Battery Level'
+        self._device_class = 'battery'
+        self._unit_of_measurement = "%"
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Subscribe to sensors events."""
+        self._device.register_async_callback(self.async_update_callback)
+
+    @callback
+    def async_update_callback(self, reason):
+        """Update the battery's state, if needed."""
+        if 'battery' in reason['attr']:
+            self.async_schedule_update_ha_state()
+
+    @property
+    def state(self):
+        """Return the state of the battery."""
+        return self._device.battery
+
+    @property
+    def name(self):
+        """Return the name of the battery."""
+        return self._name
+
+    @property
+    def device_class(self):
+        """Class of the sensor."""
+        return self._device_class
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend."""
+        return icon_for_battery_level(int(self.state))
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity."""
+        return self._unit_of_measurement
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the battery."""
+        attr = {
+            ATTR_EVENT_ID: slugify(self._device.name),
+        }
+        return attr
+
+
+class DeconzEvent(object):
+    """When you want signals instead of entities.
+
+    Stateless sensors such as remotes are expected to generate an event
+    instead of a sensor entity in hass.
+    """
+
+    def __init__(self, hass, device):
+        """Register callback that will be used for signals."""
+        self._hass = hass
+        self._device = device
+        self._event = 'deconz_{}'.format(CONF_EVENT)
+        self._id = slugify(self._device.name)
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
```

Comment:
Since this class doesn't inherit from `Entity`, you can't use this method, and you don't need to either, since you pass in `hass` when instantiating the event object. So move the registration back to init.

Reply:
Got carried away with the changes :)